### ISSUE TITLE
feat(types,html-spec): validate meta content by http-equiv value (refresh and content-type syntax)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3734]
## Summary

Validate the syntax of `content` values on `<meta http-equiv>` elements, specifically for `http-equiv="refresh"` (refresh/redirect syntax) and `http-equiv="content-type"` (MIME type with charset).

## Background

Split from #3605 (closed). The original issue included invalid test cases (`name="viewport"`, `color-scheme`, `theme-color`, `referrer` content values) — nu-html-checker does NOT validate those either. The real gap is on `http-equiv` values.

### http-equiv="refresh"

Valid syntax: `<integer>[;url=<url>]` or similar per HTML spec.

```html
<!-- Invalid -->
<meta http-equiv="refresh" content="garbage value">

<!-- Valid -->
<meta http-equiv="refresh" content="30">
<meta http-equiv="refresh" content="0;url=https://example.com/">
```

### http-equiv="content-type"

Must be `text/html; charset=<encoding>` with a valid MIME and charset.

```html
<!-- Invalid -->
<meta http-equiv="content-type" content="not a mime">

<!-- Valid -->
<meta http-equiv="content-type" content="text/html; charset=utf-8">
```

## Spec references

- https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-http-equiv-refresh
- https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-http-equiv-content-type

## Verification

nu-html-checker confirmed output (via Docker):
- `content="garbage value"` → "Bad value: Refresh value content is not well-formed"
- `content="not a mime"` → "Bad value: content-type value is not well-formed"

## Affected package

`@markuplint/types` — new validators for refresh and content-type syntax

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:59b0e97e2664, markuplint=5.0.0-rc.4, generated=2026-04-23 13:31:47 UTC)._

Filter: `meta\/refresh-.*novalid` → **5** fixtures. Verdict tally: match-error=2, match-clean=0, ml-only=0, nu-only=0, nu-over=3.

Five refresh-syntax fixtures are `nu-only`. `http-equiv="content-type"` syntax has no dedicated fixture in the current nu-validator suite — that half of the issue needs either a manual test or a new upstream fixture.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/elements/meta/refresh-empty-novalid.html` | match-error | — |
| `html/elements/meta/refresh-invalid-keyword-novalid.html` | nu-over | `nv-c57d9ac83548` |
| `html/elements/meta/refresh-missing-semicolon-novalid.html` | match-error | — |
| `html/elements/meta/refresh-missing-space-novalid.html` | nu-over | `nv-f26bf6cca5a8` |
| `html/elements/meta/refresh-quoted-url-novalid.html` | nu-over | `nv-59fd76f96902` |

Reproduce: see [`tests/external/CLAUDE.md`](https://github.com/markuplint/markuplint/blob/dev/tests/external/CLAUDE.md) "Auditing a claim against the benchmark".

**Comment by YusukeHirao** [comment]
## Docker 実測結果（2026-04-15）

### 全 9 ケース対比

| # | `content` | `http-equiv` | nu-html-checker | markuplint | Gap |
|---|-----------|-------------|----------------|-----------|-----|
| 1 | `30` | refresh | 違反なし | 違反なし | NO |
| 2 | `0;url=https://example.com/` | refresh | `Bad value "0;url=https://example.com/" ... Expected a space character, but saw "u" instead.` | 違反なし | **YES**（nu はセミコロン後のスペース要求） |
| 3 | `garbage value` | refresh | `Bad value ... Expected a digit, but saw "g" instead.` | 違反なし | **YES** |
| 4 | `abc;url=foo` | refresh | `Bad value ... Expected a digit, but saw "a" instead.` | 違反なし | **YES** |
| 5 | `text/html; charset=utf-8` | content-type | 違反なし（ただし charset 共存は別途） | 違反なし | NO |
| 6 | `not a mime` | content-type | `Bad value "not a mime" ... The legacy encoding declaration did not start with "text/html;".` | 違反なし | **YES** |
| 7 | `preferred-style` | default-style | 違反なし | 違反なし | NO |
| 8 | `IE=edge` | x-ua-compatible | 違反なし | 違反なし | NO |
| 9 | `en` | content-language | obsolete 警告（http-equiv 自体）| 「属性「http-equiv」は...のいずれかである必要があります」(`invalid-attr`) | NO（両方検出、理由違い） |

## 真のスコープ

2 つの conditional validator が必要:

1. **refresh 構文**: `<integer>` (オプションで `; url=<url>`、スペース要求あり)
2. **content-type 構文**: `text/html;` で始まり、有効な charset 宣言を含む

## 実装方針

既存の `ConditionalAttributeType` インフラ（#3189 で導入済みの link[as] 基盤）を流用:
- `<meta>` の `content` を `http-equiv` 値で分岐
- 各分岐で対応する構文 validator を適用

修正箇所:
- `@markuplint/types` — refresh / content-type 構文 validator 新規追加
- `@markuplint/html-spec` — meta 要素の condition データに ConditionalAttributeType 設定

なお `content-type` vs `charset` の document-level 共存禁止は別 Issue（#3634）で扱う。

**Comment by YusukeHirao** [comment]
Resolved by #3804 (merged).

## What landed

- `@markuplint/types`: new `HTTPEquivRefresh` and `HTTPEquivContentType` validators implementing HTML LS §4.2.5.3 (Refresh) and §4.2.5.2 (Encoding declaration).
- `@markuplint/html-spec`: `spec.meta.jsonc` now splits `meta[content]` via `ConditionalAttributeType[]` keyed on the `http-equiv` value; other `http-equiv` values and `name=` / `itemprop=` variants fall through to `Any`.
- `tests/external/snapshots/excluded-ids.json`: three nu over-detections (`refresh-invalid-keyword`, `refresh-missing-space`, `refresh-quoted-url`) recorded with verbatim HTML LS citations — per the spec those markup forms are valid, nu-validator is stricter than HTML LS.

## Fixture outcome

| Fixture | Pre | Post |
| --- | --- | --- |
| `refresh-empty-novalid` | nu-only | **match-error** |
| `refresh-missing-semicolon-novalid` | nu-only | **match-error** |
| `refresh-invalid-keyword-novalid` | nu-only | **nu-over** (spec permits) |
| `refresh-missing-space-novalid` | nu-only | **nu-over** (spec permits) |
| `refresh-quoted-url-novalid` | nu-only | **nu-over** (spec permits) |

`refresh-invalid-number-novalid` and the datatype-level `refresh-empty-novalid` in `html/datatypes/` also flipped to `match-error` as a side effect — the checker covers them too.

## `http-equiv="content-type"` scope

Implemented the shape check (`text/html; charset=<encoding-label>`), but the nu-validator fixture suite has no dedicated content-type-novalid fixture, so bench cannot regress-test that half. Confirmed via integration tests in `invalid-attr/index.spec.ts` (`[invalid-attr-issue-3734-006]..-008]`) and unit tests in `check-http-equiv-content-type.spec.ts`.

## Bench delta

- match-error: 2058 → 2063 (+5)
- nu-only: 1455 → 1447 (−8)
- nu-over: 25 → 28 (+3)
- overall match rate: 54.8% → 54.9%
